added facade for show patterns.

svirfneblyn opened the pull request.

svirfneblyn committed:
added facade for show patterns.

svirfneblyn committed:
added facade for show patterns.

rabtsauyv commented on src/main/java/adapter/BridgeEnum.java:
```diff
@@ -19,15 +15,11 @@ public CrossingImpl getCross() {
-
     public abstract CrossingImpl getCross();
-
     private String BridgeType;
```
code convention: 
1) field name should begin with lowercase letter
2) order: static fields, instance fields, constructors, methods

cslisenka replied:
согласен

rabtsauyv commented on src/main/java/composite/DirectoryEntity.java:
```diff
@@ -3,47 +3,51 @@
 
     private String directoyName;
     private long dirSize;
     private List<FileEntity> fileEntityList = new ArrayList();
```
`List<FileEntity>` - допускает всего 1 уровень вложенности, т.е. внутри твоей папки могут храниться только файлы, вложенных папок быть не может;
вероятно, должно быть `List<FileSystemEntity>`

cslisenka replied:
сделать final

rabtsauyv commented on src/main/java/composite/DirectoryEntity.java:
```diff
@@ -3,47 +3,51 @@
 public class DirectoryEntity implements FileSystemEntity {
 
     private String directoyName;
     private long dirSize;
```
размер папки стоит сделать вычисляемым, в зависимости от размера содержащихся в ней файлов;
идея паттерна - чтобы при вызове переопределенного метода из интерфейса, напр. `fileSystemEntitySize`, на одном объекте (файле) - была своя логика, а при вызове на коллекции объектов (папке с файлами) - своя

cslisenka replied:
Проверить где используется dirSize.

cslisenka commented on src/main/java/composite/DirectoryEntity.java:
```diff
@@ -3,47 +3,51 @@
     }
+
     @Override
     public void fileSystemEntityName(String fn) {
```
Что делает этот метод? Просто печатает в консоль?
Название должно быть глаголом. Если ты хочешь задать имя или размер, то имя метода стоит сделать setFileSystemEntity...

cslisenka commented on src/main/java/composite/FileEntity.java:
```diff
@@ -20,14 +18,15 @@ private long getFileSize() {
     }
 
     public void fileSystemEntityName(String fn) {
         this.fileName=fn ;
```
пропущены пробелы до и после =

cslisenka replied:
code style
более осмысленное имя атрибута

cslisenka commented on src/main/java/decorator/BridgeDecorator.java:
```diff
@@ -0,0 +1,19 @@
+    protected ICrossing cr;
+    public BridgeDecorator(ICrossing cr) {
+        this.cr = cr;
+    }
```
После метода должна быть одна пустая строка

cslisenka commented on src/main/java/decorator/BridgeDecorator.java:
```diff
@@ -0,0 +1,19 @@
+
+/*decorator*/
+public class BridgeDecorator implements ICrossing {
+    protected ICrossing cr;
```
перед методом должна быть одна пустая строка

cslisenka commented on src/main/java/facade/PatternDemonstrator.java:
```diff
@@ -0,0 +1,59 @@
+        FlyweightStylesFactory flyweight =  FlyweightStylesFactory.getInstance();
+        flyweight.getBlogerFlyweight("horror").getStyleForBlog("coolBlg");
+        flyweight.getBlogerFlyweight("funny").getStyleForBlog("funnylBlg");
+
```
пустая строка лишняя

cslisenka commented on src/main/java/facade/PatternDemonstrator.java:
```diff
@@ -0,0 +1,59 @@
+    }
+    public void showFlyweightPat() {
+        printPatternType("FLYWEIGHT");
+        FlyweightStylesFactory flyweight =  FlyweightStylesFactory.getInstance();
```
после = должен быть только 1 пробел

cslisenka commented on src/main/java/flyweight/FliweightFunnyStyleBlog.java:
```diff
@@ -1,10 +1,7 @@
 public class FliweightFunnyStyleBlog implements BloggerStyleFlyweight {
-
     public void getStyleForBlog(String blogName) {
     System.out.println("Selected funny style for " + blogName);
```
пропущена табуляция

cslisenka commented on src/main/java/adapter/CompositeAndBridgeAdapter.java:
```diff
@@ -1,10 +1,7 @@
- */
+/*adapter*/
 public interface CompositeAndBridgeAdapter {
     public void buildBridge(BridgeEnum typeToCross);
```
1. Я в замешательстве. Какая связь между мостами и file name scanner? Меня смущает что вещи абсолютно разной природы находятся в одном адаптере :)
2. Почему методы называются build и get, но ничего не возвращают?

cslisenka commented on src/main/java/adapter/CompositeAndBridgeAdapter.java:
```diff
@@ -1,10 +1,7 @@
- */
+/*adapter*/
 public interface CompositeAndBridgeAdapter {
     public void buildBridge(BridgeEnum typeToCross);
```
Нужно что-то вернуть судя из названия методов
Название build не логично т.к. bridge уже создан, он в enum-е, тут скорее уместно printBridge

cslisenka replied:
Адаптер должен связывать две системы, одна из них вызывает другую. Обе системы имеют разный интерфейс.

cslisenka commented on src/main/java/bridge/icrossing/ICrossing.java:
```diff
@@ -1,8 +1,6 @@
- */
+/*bridge*/
 public interface ICrossing {
     public void deploy();
```
В интерфейсах pucliс можно не писать

cslisenka commented on src/main/java/bridge/CrossingImpl.java:
```diff
@@ -2,12 +2,9 @@
- * Created by Ihar_Rubanovich on 3/7/2017.
- */
+/*bridge*/
 public abstract class CrossingImpl {
```
Можно было бы заимплементить ICrossing

cslisenka replied:
Если класс называется CrossingImpl, то можно предположить что он должен имплементить ICrossing.
Тут же используется агрегация. Возможно стоит переименовать.

cslisenka commented on src/main/java/bridge/CrossingImpl.java:
```diff
@@ -2,12 +2,9 @@
- */
+/*bridge*/
 public abstract class CrossingImpl {
     public ICrossing iCrossing;
```
делай такие атрибуты final (если инициализируется только в конструкторе)

cslisenka commented on src/main/java/composite/DirectoryEntity.java:
```diff
@@ -3,47 +3,51 @@
+/*composite*/
 public class DirectoryEntity implements FileSystemEntity {
 
     private String directoyName;
```
final

cslisenka commented on src/main/java/composite/DirectoryEntity.java:
```diff
@@ -3,47 +3,51 @@
         return fileEntityList;
     }
+
     @Override
```
Пересмотреть надобность методов
Либо переделать либо убрать

cslisenka commented on src/main/java/composite/FSManager.java:
```diff
@@ -4,36 +4,29 @@
 public class FSManager {
-    public  void compositDemo(String volume) {
+
+    public void compositDemo(String volume) {
```
path

cslisenka commented on src/main/java/composite/FSManager.java:
```diff
@@ -4,36 +4,29 @@
         List<DirectoryEntity> dirList = new ArrayList<>();
         File[] allFiles = listFile.listFiles();
-
         assert allFiles != null;
```
никогда не используй такие асерты

cslisenka commented on src/main/java/composite/FSManager.java:
```diff
@@ -4,36 +4,29 @@
                 System.out.println("Directory  " + file.getName() + " added ");
             }
         }
         System.out.println("File System Ierarchy readed ");
```
Лучше возвращать и принтить вне класса с бизнес-логикой

cslisenka commented on src/main/java/composite/FSManager.java:
```diff
@@ -4,36 +4,29 @@
         dirList.forEach(DirectoryEntity::print);
     }
-
     private static DirectoryEntity addFiles(File[] filesInDir, DirectoryEntity de) {
```
я не делаю статики без необходимости

cslisenka commented on src/main/java/composite/FileEntity.java:
```diff
@@ -1,8 +1,6 @@
+/*composite*/
 public class FileEntity implements FileSystemEntity {
 
     private String fileName;
```
через конструктор и final

cslisenka commented on src/main/java/composite/FileSystemEntity.java:
```diff
@@ -1,8 +1,6 @@
- */
+/*composite*/
 public interface FileSystemEntity {
     public void fileSystemEntityName(String fn);
```
наверное get...

cslisenka commented on src/main/java/facade/PatternDemonstrator.java:
```diff
@@ -0,0 +1,59 @@
+import static flyweight.Constants.PATH_TO_SCAN;
+
+/*facade*/
+public class PatternDemonstrator {
```
это скорее god object
фасад в моём понимании не должен делать бизнес-логику

cslisenka commented on src/main/java/flyweight/FlyweightStylesFactory.java:
```diff
@@ -6,9 +6,7 @@
+/*flyweight*/
 public class FlyweightStylesFactory {
     private static volatile FlyweightStylesFactory flyweightStylesFactory;
     private Map<String, BloggerStyleFlyweight> bStyleFlyweightMap;
```
ConcurrentHashMap
putIfAbsent

svirfneblyn committed:
fixed codestyle and facade

svirfneblyn committed:
исправлено согласно замечаниям.
code style:
- имена методов
- имена классов
- пробелы и табуляции

паттер адаптер
паттерн композит
паттерн фасад .
